### PR TITLE
add remove_duplicate_whitspace function

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
Hey team. Please let me know if you're interested in this function. If so, I can try to add some tests (if desired/needed).

File: lib/puppet/parser/functions/remove_duplicate_whitespace.rb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    raise(Puppet::ParseError, "remove_duplicate_whitespace(): Wrong number of arguments given (#{arguments.size} for 1)") if arguments.empty?
+
+    value = arguments[0]
+
+    unless value.is_a?(Array) || value.is_a?(String)
+      raise(Puppet::ParseError, 'remove_duplicate_whitespace(): Requires either array or string to work with')
+    end
+
+    result = if value.is_a?(Array)
+               value.map { |i| i.is_a?(String) ? i.gsub(/\s+/, ' ') : i }
```
Comment (ekohl) [inline lib/puppet/parser/functions/remove_duplicate_whitespace.rb]: How about keeping the original whitespace char. Something like ```gsub(/(\s)\s+/, '\1')```

Comment (bastelfreak) [reply lib/puppet/parser/functions/remove_duplicate_whitespace.rb]: Would that make a huge difference?

Comment (ekohl) [reply lib/puppet/parser/functions/remove_duplicate_whitespace.rb]: I'm wondering if people like using tabs or perhaps tabs are needed somehow?

File: lib/puppet/parser/functions/remove_duplicate_whitespace.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#  remove_duplicate_whitespace.rb
+#
+module Puppet::Parser::Functions
+  newfunction(:remove_duplicate_whitespace, :type => :rvalue, :doc => <<-DOC
+    This function replaces one or multiple whitespaces with a single one from a string or from
+    every string inside an array.
+
+    *Examples:*
+
+        strip("    aaa   bla")
```
Comment (binford2k) [inline lib/puppet/parser/functions/remove_duplicate_whitespace.rb]: The name doesn't match the docs

**Comment by bastelfreak** [comment]
I ruined my git history in this feature branch and cleaned it up.

File: lib/puppet/parser/functions/remove_duplicate_whitespace.rb (hunk 2 above)
Comment (bastelfreak) [reply lib/puppet/parser/functions/remove_duplicate_whitespace.rb]: fixed

**Comment by hlindberg** [comment]
I think this could be a useful addition to the String formatting that is built into puppet - see [specification for String(String)](https://github.com/puppetlabs/puppet-specifications/blob/master/language/types_values_variables.md#string-to-string) - it has a %t format that trims left/right, but none that compacts internal whitespace. A %T could trim all spaces - so users can write `$result = String($input, '%T')`.

**Comment by hlindberg** [comment]
There is already a `squeeze` function that can do this. Closing this.

**Pull request closed without merging**